pkg/localmetrics/localmetrics.go

```diff
@@ -15,176 +15,152 @@
-	MetricsList = []prometheus.Collector{
-		MetricTotalAWSAccounts,
-		MetricTotalAccountCRs,
-		MetricTotalAccountCRsUnclaimed,
-		MetricTotalAccountCRsClaimed,
-		MetricTotalAccountCRsFailed,
-		MetricTotalAccountClaimCRs,
-		MetricTotalAccountCRsReady,
-		MetricPoolSizeVsUnclaimed,
-		MetricTotalAccountPendingVerification,
-		MetricTotalAccountReusedAvailable,
-		MetricTotalAccountReuseFailed,
-	}
+const (
+	operatorName = "aws-account-operator"
 )
 
-// UpdateAccountCRUnclaimedMetric updates the unclaimed account metric
-func UpdateAccountCRUnclaimedMetric(accountList awsv1alpha1.AccountList, wg *sync.WaitGroup) {
+var log = logf.Log.WithName("metrics-collector")
 
-	wg.Add(1)
-	unclaimedAccountCount := 0
+type MetricsCollector struct {
+	store                  cache.Cache
+	awsAccounts            prometheus.Gauge
+	accounts               *prometheus.GaugeVec
+	accountClaims          *prometheus.GaugeVec
+	accountReusedAvailable *prometheus.GaugeVec
+	accountPoolSize        *prometheus.GaugeVec
+}
 
-	for _, account := range accountList.Items {
-		if account.Status.Claimed == false && account.Status.Reused == false {
-			if account.Status.State == "Ready" || account.Status.State == string(awsv1alpha1.AccountPendingVerification) {
-				unclaimedAccountCount++
-			}
-		}
+func NewMetricsCollector(store cache.Cache) *MetricsCollector {
+	return &MetricsCollector{
+		store: store,
+		awsAccounts: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name:        "aws_account_operator_aws_accounts",
+			Help:        "Report how many accounts have been created in AWS org",
+			ConstLabels: prometheus.Labels{"name": operatorName},
+		}),
+		accounts: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name:        "aws_account_operator_account_crs",
+			Help:        "Report how many account CRs have been created",
+			ConstLabels: prometheus.Labels{"name": operatorName},
+		}, []string{"claimed", "reused", "ccs", "state"}),
+		accountClaims: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name:        "aws_account_operator_account_claim_crs",
+			Help:        "Report how many account claim CRs have been created",
+			ConstLabels: prometheus.Labels{"name": operatorName},
+		}, []string{"state"}),
+		accountReusedAvailable: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name:        "aws_account_operator_aws_accounts_reusable",
+			Help:        "Report the number of reused accounts available for claiming grouped by legal ID",
+			ConstLabels: prometheus.Labels{"name": operatorName},
+		}, []string{"ccs", "legal_id"}),
+
+		// pool_name is not a good label because it may cause
+		// high cardinality. But in our use case it is okay
+		// since we only has one account pool in the cluster.
+		accountPoolSize: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name:        "aws_account_operator_account_pool_size",
+			Help:        "Report the size of account pool CR",
+			ConstLabels: prometheus.Labels{"name": operatorName},
+		}, []string{"namespace", "pool_name"}),
```

Comment:
In practice, we will only ever have one pool due to the design of hive. The cluster name gets appended to the metric in the app-sre prometheus (i.e. `hive-stage`). I'm pretty sure we have the same pool name across all our environments anyway so I'm not sure if this label is useful

Reply:
You are totally right. But I think the design of `account pool cr` is quite weird, if we can only have one in the cluster, then why do we need that cr?

Reply:
The CR was added when we had a different design. This is just the way its worked out.

Reply:
I think it is okay to keep the `pool_name` label. Though it is not very useful, it won't cause cardinality problems. What do you think?